sim/rogue/items_sets_pve.go

```diff
@@ -213,3 +215,110 @@ var ItemSetNightSlayerBattlearmor = core.NewItemSet(core.ItemSet{
+							return spell != nil && spell.Cost != nil && spell.Cost.CostType() == core.CostTypeEnergy
+						},
+					)
+				},
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					core.Each(affectedSpells, func(spell *core.Spell) { spell.Cost.Multiplier -= 100 })
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					core.Each(affectedSpells, func(spell *core.Spell) { spell.Cost.Multiplier += 100 })
+				},
+				OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
+					if aura.RemainingDuration(sim) == aura.Duration || spell.DefaultCast.Cost == 0 {
+						return
+					}
+					aura.Deactivate(sim)
+				},
+			})
+
+			core.MakePermanent(rogue.RegisterAura(core.Aura{
+				Label:    "S03 - Item - T2 - Rogue - Damage 2P Bonus",
+				ActionID: core.ActionID{SpellID: 467708},
+				OnSpellHitDealt: func(_ *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if !result.Landed() || !spell.ProcMask.Matches(core.ProcMaskMeleeOrRangedSpecial) {
+						return
+					}
+					switch spell {
+					case rogue.Ambush, rogue.Garrote:
+						aura.Activate(sim)
+					}
+				},
+			}))
+		},
+		// Increases main hand weapon damage by 10%
+		4: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.OnSpellRegistered(func(spell *core.Spell) {
+				if spell.ProcMask.Matches(core.ProcMaskMeleeMH) {
+					spell.DamageMultiplier *= 1.10
+				}
+			})
+		},
+			// Reduces cooldown on vanish by 2.5 minutes
+		6: func(agent core.Agent) {
+			// Implemented in Vanish.go
+		},
+	},
+})
+
+var ItemSetBloodfangBattlearmor = core.NewItemSet(core.ItemSet{
+	Name: "Bloodfang Battlearmor",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Your Rolling with the Punches now also activates every time you gain a combo point.
+		2: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			if !rogue.HasRune(proto.RogueRune_RuneRollingWithThePunches) {
+				return
+			}
+
+//			rogue.OnComboPointsGained(func(sim *core.Simulation) {
+//				if rogue.RollingWithThePunchesProcAura.GetStacks() == 5 {
+//					rogue.RollingWithThePunchesProcAura.Refresh(sim)
+//				} else if rogue.RollingWithThePunchesProcAura.IsActive() {
+//					rogue.RollingWithThePunchesProcAura.AddStack(sim)
+//				} else {
+//					rogue.RollingWithThePunchesProcAura.Activate(sim)
+//					rogue.RollingWithThePunchesProcAura.AddStack(sim)
+//				}
+//			})
+		},
+		// Your Rolling with the Punches also grants you 20% increased Armor from items per stack.
+		4: func(agent core.Agent) {
+			//Implemented in runes.go RollingwiththePunches function TO-DO
+		},
+			// The cooldown on your Main Gauche resets every time your target Dodges or Parries.
+		6: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			
+			core.MakePermanent(rogue.RegisterAura(core.Aura{
+				Label:    "S03 - Item - T2 - Rogue - Tank 6P Bonus",
+				ActionID: core.ActionID{SpellID: 467803},
```

Comment:
Unless you purposefully want this to show up all the time in the timeline you could remove this

Reply:
Bump

Reply:
done